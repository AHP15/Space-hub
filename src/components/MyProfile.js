import '../styles/MyProfile.css';
import React from 'react';
import { useSelector } from 'react-redux';

// eslint-disable-next-line arrow-body-style
const MyProfile = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  return (
    <div className="profile">
      <div className="missions">
        <h2>My Messions</h2>
      </div>
      <div className="rockets">
        <h2>My Rockets</h2>
        {reservedRockets.length > 0 ? <ul>{reservedRockets.map((rocket) => <li key={rocket.id}>{rocket.rocket_name}</li>)}</ul> : ''}
      </div>
    </div>
  );
};
export default MyProfile;
