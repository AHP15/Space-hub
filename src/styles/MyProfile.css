.profile {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  my-profile-filters
  padding: 60px 100px;
  gap: 30px;
}

.profile div {
  flex: 1;
}

my-profile-filters
.profile div h2 {
  margin-bottom: 20px;
}

.mission_name {
  padding: 20px;
  border: 1px solid lightgray;
  margin: 5px;
}

.rockets h2 {
  font-weight: 500;
}

.rockets ul {
  list-style: none;
  border: 1px solid #acacac;
  margin-top: 15px;
}

.rockets ul li {
  padding: 15px 0 20px 20px;
}

.rockets ul li:not(:last-child) {
  border-bottom: 1px solid #acacac;
}
