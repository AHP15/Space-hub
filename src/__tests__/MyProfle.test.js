import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';

import MyProfile from '../components/MyProfile';
import Store from '../slices/store';

it('testing MyProfile', () => {
  const tree = renderer.create(<Provider store={Store}><MyProfile /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
