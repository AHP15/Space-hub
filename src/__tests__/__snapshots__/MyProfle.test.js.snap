// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MyProfile 1`] = `
<div
  className="profile"
>
  <div
    className="missions"
  >
    <h2>
      My Messions
    </h2>
    <p
      className="mission_name"
    >
      mission name
    </p>
  </div>
  <div
    className="rockets"
  >
    <h2>
      My Rockets
    </h2>
    <ul>
      <li>
        mission name
      </li>
    </ul>
  </div>
</div>
`;
